src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,76 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String model;
+    private Engine clone;
+    private String color;
+    private int age;
+    private int maxSpeed;
+    private Engine engine;
 
+    private Car(CarBuilder builder) {
+        this.model = builder.model;
+        this.color = builder.color;
+        this.age = builder.age;
+        this.maxSpeed = builder.maxSpeed;
+        this.engine = builder.engine;
+
+    }
+
+    public Car(String model, Engine clone) {
+        this.model = model;
+        this.clone = clone;
+    }
+
+    public static class CarBuilder {
+        private String model;
+        private String color;
+        private int age;
+        private int maxSpeed;
+        private Engine engine;
+
+        public CarBuilder(String model, Engine engine) {
+            this.model = model;
+            this.engine = engine;
+        }
+
+        public CarBuilder setColor(String color) {
+            this.color = color;
+            return this;
+        }
+
+        public CarBuilder setModel(String model) {
+            this.model = model;
+            return this;
+        }
+
+        public CarBuilder setAge(int age) {
+            this.age = age;
+            return this;
+        }
+
+        public CarBuilder setMaxSpeed(int maxSpeed) {
+            this.maxSpeed = maxSpeed;
+            return this;
+        }
+
+        public CarBuilder setFuelEngine(Engine engine) {
+            this.engine = engine;
+            return this;
+        }
+
+        public Car build() {
+            return new Car(this);
+        }
+    }
+
+    @Override
+    public Car clone() {
```

Comment:
```suggestion
    Car clonedCar = (Car) super.clone();
    clonedCar.setEngine(engine.clone());
    return car;
```
same her but manual set of engine